### DISCUSSION TITLE
clarification on topic operator and PV

### DISCUSSION THREAD

**Discussion by shreyasarani23** [strimzi/strimzi-kafka-operator#4960] [General]
Hi @scholzj I have one doubt/confusion say we have kafka cluster configured with persistent volume and below is it's configuration
 storage:
      type: persistent-claim
      size: 1Gi
      class: kafka-storage-class
      deleteClaim: true
  entityOperator:
    userOperator: {}
    topicOperator: {}
In the above configuration when should we use deleteClaim:true and when should we use deleteClaim:false?
Here are the things for which I need clarification.
1. when I delete my kafka cluster using kubectl delete -f kafka.yaml -n kafka command and redeploy it again kubectl apply command does the topic operator delete all the topics?
2.  when I delete my kafka cluster and while redeploying it do I need to follow the steps mentioned in the link https://strimzi.io/docs/operators/in-development/using.html#cluster-recovery-volume_str? even though my namespace is not deleted, kubernetes cluster running and have used deleteClaim:false ??
OR just redeploy my kafka cluster with kubectl apply command?
please can you clarify?

**Comment by scholzj** [comment]
`deleteClaim: true` is something you can you for example in development. It makes sure that when the Kafka CR is deleted, the storage will be deleted as well. This is handy for example when trying stuff during development, where it makes sure the volumes are deleted and you do not have t pay for them to some cloud provider if you forget to delete them manually.

The topic operator does a 3 way reconciliation. Between:
* The KafkaTopic custom resources
* The topics in Kafka
* Its own state

When you have existing cluster with existing topics, they will exist in all 3 places. But if you during the recovery don't recover all of them to the original state - in this case if the topic resources will be missing in Kubernetes but will be present in the other two places, the TO will see that as the topics being deleted and will proceed and delete them from Kafka. If the topics are not present in both the internal state and in Kubernetes, the TO will see it as new topics being added in Kafka and will proceed and create the KafkaTopic resources.

**Reply by shreyasarani23** [reply]
U mean to say that topic operator doesn't delete the topics when the kafka cluster is redeployed right?
I didn't understand the reconciliation concept can u share me the link where they have explained about this 3 way reconciliation. And what do u mean by kafkaTopic custom resources?

Can you clarify more about this point?
when I delete my kafka cluster and while redeploying it do I need to follow the steps mentioned in the link https://strimzi.io/docs/operators/in-development/using.html#cluster-recovery-volume_str? even though my namespace is not deleted, kubernetes cluster running and have used deleteClaim:false ??
OR just redeploy my kafka cluster with kubectl apply command?

**Reply by scholzj** [reply]
There is no general answer. It depends on what you deleted from the namespace or did not deleted from the namespace. If all you deleted is the Kafka custom resource, you do not need to follow it. But keep in mind that this is a disaster recovery procedure - not something you are supposed to do every day for the evening.

**Reply by shreyasarani23** [reply]
Ok got it @scholzj
